test/e2e/kube-ovn/ipam/ipam.go

```diff
@@ -509,4 +509,57 @@ var _ = framework.Describe("[group:ipam]", func() {
 		deploy = deployClient.PatchSync(deploy, patchedDeploy)
 		checkFn()
 	})
+
+	framework.ConformanceIt("should allocate right IPs for the statefulset when there are multiple IP Pools added to its namespace", func() {
+		f.SkipVersionPriorTo(1, 14, "Multiple IP Pools per namespace support was introduced in v1.14")
+		replicas := 1
+		ipsCount := 12
+		testStsName := "test-statefulset" + framework.RandomSuffix()
+		testSubnetName := "ip-pool-subnet2" + framework.RandomSuffix()
+		testIPPool1Name := "ip-pool1" + framework.RandomSuffix()
+		testIPPool2Name := "ip-pool2" + framework.RandomSuffix()
+
+		ginkgo.By("Creating a new subnet " + testSubnetName)
+		testCidr := framework.RandomCIDR(f.ClusterIPFamily)
+		testSubnet := framework.MakeSubnet(testSubnetName, "", testCidr, "", "", "", nil, nil, []string{namespaceName})
+		subnetClient.CreateSync(testSubnet)
+
+		ginkgo.By("Creating IPPool resources ")
+		ipsRange1 := framework.RandomIPPool(cidr, ipsCount)
+		ipsRange2 := framework.RandomIPPool(testCidr, ipsCount)
+		ippool1 := framework.MakeIPPool(testIPPool1Name, subnetName, ipsRange1, []string{namespaceName})
+		ippool2 := framework.MakeIPPool(testIPPool2Name, testSubnetName, ipsRange2, []string{namespaceName})
+		ippoolClient.CreateSync(ippool1)
+		ippoolClient.CreateSync(ippool2)
+
+		ginkgo.By("Creating statefulset " + testStsName + " with logical switch annotation and no ippool annotation")
+		labels := map[string]string{"app": testStsName}
+		sts := framework.MakeStatefulSet(testStsName, testStsName, int32(replicas), labels, framework.PauseImage)
+		sts.Spec.Template.Annotations = map[string]string{util.LogicalSwitchAnnotation: subnetName}
+		sts = stsClient.CreateSync(sts)
+
+		ginkgo.By("Getting pods for statefulset " + testStsName)
+		pods := stsClient.GetPods(sts)
+		framework.ExpectHaveLen(pods.Items, replicas)
+
+		for _, pod := range pods.Items {
+			framework.ExpectHaveKeyWithValue(pod.Annotations, util.AllocatedAnnotation, "true")
+			framework.ExpectHaveKeyWithValue(pod.Annotations, util.CidrAnnotation, subnet.Spec.CIDRBlock)
+			framework.ExpectHaveKeyWithValue(pod.Annotations, util.GatewayAnnotation, subnet.Spec.Gateway)
+			framework.ExpectHaveKeyWithValue(pod.Annotations, util.LogicalSwitchAnnotation, subnetName)
+			framework.ExpectIPInCIDR(pod.Annotations[util.IPAddressAnnotation], subnet.Spec.CIDRBlock)
+			framework.ExpectMAC(pod.Annotations[util.MacAddressAnnotation])
+			framework.ExpectHaveKeyWithValue(pod.Annotations, util.RoutedAnnotation, "true")
+		}
+
+		ginkgo.By("Deleting statefulset " + testStsName)
+		stsClient.DeleteSync(testStsName)
+
+		ginkgo.By("Deleting ippools")
+		ippoolClient.DeleteSync(testIPPool1Name)
+		ippoolClient.DeleteSync(testIPPool2Name)
+
+		ginkgo.By("Deleting subnet " + testSubnetName)
+		subnetClient.DeleteSync(testSubnetName)
```

Comment:
We should use `ginkgo.DeferCleanup()` to clean up resources created by test cases.

Reply:
Not sure why, I'm getting a timeout when trying to use DeferCleanup
```bash
 msg: "expected subnet subnet-596431823 to not be found
    err: <framework.FailureError>{
              msg: "Timed out after 120.002s.\nExpected\n    <*v1.Subnet
 ```

Reply:
Another method is to move the resource name variable definitions into the `framework.Describe()` argument function, like what we do for `subnetName`.